fastlane/lib/fastlane/actions/register_device.rb

```diff
@@ -19,8 +17,13 @@ def self.run(params)
         Spaceship.login(credentials.user, credentials.password)
         Spaceship.select_team
 
-        UI.user_error!("Passed invalid UDID: #{udid} for device: #{name}") unless UDID_REGEXP =~ udid
-        Spaceship::Device.create!(name: name, udid: udid)
+        begin
+          Spaceship::Device.create!(name: name, udid: udid)
+        rescue => ex
+          UI.error("Failed to register new device")
```

Comment:
What's the difference between `UI.error` and `UI.user_error!` that was used before?

Reply:
- `UI.error` will print out message formatted as errors to the console (typically red)
- `UI.user_error!` will also print out messages formatted in errors but will also raise an error (which is what the `!` means here` as this will halt execution)

Reply:
Returning `nil` achieves the same as raising an error?